ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/admin/config_management_service.proto

```diff
@@ -0,0 +1,77 @@
+
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.digitalasset.ledger.api.v1.admin;
+
+option java_outer_classname = "ConfigManagementServiceOuterClass";
+option java_package = "com.digitalasset.ledger.api.v1.admin";
+option csharp_namespace = "Com.DigitalAsset.Ledger.Api.V1.Admin";
+
+import "google/protobuf/duration.proto";
+
+// Status: experimental interface, will change before it is deemed production
+// ready
+
+// Ledger configuration management service provides methods for the ledger administrator
+// to change the current ledger configuration. The services provides methods to modify
+// different aspects of the configuration.
+service ConfigManagementService {
+  // Return the currently active time model and the current configuration generation.
+  rpc GetTimeModel (GetTimeModelRequest) returns (GetTimeModelResponse);
+
+  // Set the ledger time model.
+  // In case of failure this method responds with:
+  // - INVALID_ARGUMENT if arguments are invalid, or the provided configuration generation
+  //   does not match the current active configuration generation. The caller is expected
+  //   to retry by again fetching current time model using 'GetTimeModel', applying changes
+  //   and resubmitting.
+  // - UNIMPLEMENTED if this method is not supported by the backing ledger.
+  rpc SetTimeModel (SetTimeModelRequest) returns (SetTimeModelResponse);
+
+  // FIXME(JM): Support retrieving the complete configuration encoded using ledger_configuration.proto?
+  // FIXME(JM): Support streaming configuration?
+}
+
+// FIXME(JM): ../ledger_configuration_service.proto contains "LedgerConfiguration" that
+// has some time model parameters. It isn't a 1-1 match though. We might want to reconcile these.
+
+// FIXME(JM): Complete copy from ledger_configuration.proto. We likely still want to evolve these
+// separately as the ledger_configuration.proto has to maintain full backwards compatibility.
+message TimeModel {
+  // The expected minimum latency of a transaction.
+  google.protobuf.Duration min_transaction_latency = 1;
+
+  // The maximum allowed clock skew between the ledger and clients.
+  google.protobuf.Duration max_clock_skew = 2;
+
+  // The maximum allowed time to live for a transaction.
+  // Must be greater than the derived minimum time to live.
+  google.protobuf.Duration max_ttl = 3;
+}
+
+message GetTimeModelRequest {
+}
+
+message GetTimeModelResponse {
+  // The current configuration generation. The generation is a monotonically increasing
+  // integer that is incremented on each change. Used when setting the time model.
+  int64 configuration_generation = 1;
+
+  // The current ledger time model.
+  TimeModel time_model = 2;
+}
+
+message SetTimeModelRequest {
+  // The current configuration generation which we're submitting the change against.
```

Comment:
Please add also command_id, to allow for duplicate request filtering in case of a retry after a crash

Reply:
are we switching everything to "command_id" now?

Reply:
Added it as submission id.

Reply:
subbmission_id makes more sense, thank you